torch/_dynamo/variables/builtin.py

```diff
@@ -1465,12 +1475,16 @@ def _unimplemented():
         if isinstance(left, UserDefinedObjectVariable) and isinstance(
             right, UserDefinedObjectVariable
         ):
-            return ConstantVariable.create(op(left.value, right.value))
-
-        if op.__name__ == "is_":
-            # If the two objects are of different type, we can safely return False
-            if type(left) is not type(right):
-                return ConstantVariable.create(False)
+            ret = op(left.value, right.value)
+            if op is operator.is_ and not ret:
+                guard = left.source.make_guard(
+                    functools.partial(
+                        GuardBuilder.UNALIASED_INPUT, source_b=right.source
+                    )
+                )
+                return ConstantVariable.create(ret, guards=set({guard}))
```

Comment:
Btw @lezcano you might need to do something similar for https://github.com/pytorch/pytorch/pull/111196

If this gets merged first perhaps you could use this infra.

Basically, we need to guard that `obj, obj2` are unaliased whenever `obj is not obj2` as they are not true constants and not guarded adequately. See: https://github.com/pytorch/pytorch/issues/111562

Reply:
Perhaps I can also try to handle 
```
BuiltinVariable,
EnumVariable,
```

In this PR. Basically they ought to be treated the same way as ConstantVariable methinks.

Reply:
Turns out that we do not intend to support UserDefinedObjects - there is too much risk. Hence, let's just graph break,

Reply:
Currently, we don't graph break. But now we need to. Since this may be breaking, move into new PR